Docs/BazelForiOSDevelopers.md

```diff
@@ -0,0 +1,370 @@
+
+Bazel makes it easy to manage build configuration in a way that results are
+functional and reproducible. In addition to producing an iOS application, Bazel
+makes it easy to automate many other kinds tasks for example, code generating a
+thrift schema, generating an Xcode project, and pushing a docker container.
+Having the ability to optimize the build system can significantly improve the
+developer experience.
+
+Compared to other mainstream build systems, Bazel is strict about inputs and
+outputs. This characteristic makes it reproducible and well suited for
+distributed cloud builds and caching. _To learn about how Bazel compares to
+other build systems, Microsoft's paper [Build Systems à la
+Carte](https://www.microsoft.com/en-us/research/uploads/prod/2018/03/build-systems.pdf),
+compares popular build systems._
+
+## Introduction to Bazel projects
+
+The root of a Bazel project contains two human readable files:
+
+The `BUILD` file - these files define what targets are inside of a project.
+Applications, extensions, static libraries, and frameworks are all declared in
+BUILD files.
+
+The `WORKSPACE` file - this file is about getting files and dependencies from
+[outside the world into
+Bazel](https://docs.bazel.build/versions/master/be/workspace.html).  Simply
+put, external-to-bazel dependencies are put here. 
+
+_The closest notion to BUILD files in the Xcode world, is the machine readble
+`xcodeproj` files managed by Xcode._
+
+## WORKSPACE configuration and setting up rules_apple
+
+For building iOS applications, most iOS developers use the rule set
+`rules_apple`. `rules_apple` contains key rules for iOS development which
+includes rules to build applications, unit tests, and more.
+
+Head to [rules_apple](https://github.com/bazelbuild/rules_apple) and follow the
+latest instructions to get setup. `rules_apple` provides instructions to add
+dependencies into the `WORKSPACE` file. Simply paste the lines from
+`rules_apple`'s README into the `WORKSPACE` file
+
+```
+# /path/to/myproject/WORKSPACE
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+
+git_repository(
+    name = "build_bazel_rules_apple",
+    remote = "https://github.com/bazelbuild/rules_apple.git",
+    commit = "[SOME_HASH_VALUE]",
+)
+```
+
+The first line of code calls the function, `load`. The load function imports
+symbols from a `.bzl` file.  In Objective-C or swift, it's like importing a
+header file.
+
+```
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+```
+_load `git_repository` from the internal [bazel tools](https://github.com/bazelbuild/bazel/issues/4301) repository_
+
+The next line of code calls `git_repository` defines `build_bazel_rules_apple`
+from the git repository, `https://github.com/bazelbuild/rules_apple` for a
+given `commit`. 
+
+_`git_repository` documentation https://docs.bazel.build/versions/2.0.0/repo/git.html_
+
+### BUILD files
+
+`BUILD` files are where all the targets are defined.
```

Comment:
briefly list off examples of targets (application, library, test, ...)

Reply:
Done